### PR TITLE
Reduce Process test time

### PR DESCRIPTION
**Pull request by stephentoub**
Fix a test that was taking 30 seconds, and make another test outerloop.

cc: @danmosemsft

### CODE REVIEW

File: src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 1]
```diff
@@ -1877,6 +1877,7 @@ public void Process_StartInvalidNamesTest()
             Assert.Throws<Win32Exception>(() => Process.Start("exe", string.Empty, new SecureString(), "thisDomain"));
         }
 
+        [OuterLoop("May take many seconds the first time it's run")]
```
Comment (danmosemsft) [inline src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs:1880]: Why?

**Review by danmoseley — APPROVED** [review]
How did I not think of doing it this way?

File: src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs:1880]: Not entirely sure.  I'm assuming it has something to do with my domain, or VPN, or something.  When I first run the test suite, it takes almost 30 seconds for this one test.  If I then run it again, it's less than a second.  I've not profiled to see where the time is being spent.

**Comment by danmosemsft** [comment]
Strictly, it would be possible to implement p.Kill() such that it caused the wait (using method on the same process it) to complete (as tested here) but independently terminating the process (or allowing it to exit) did not. I guess that's what I was thinking.

**Comment by stephentoub** [comment]
> Strictly, it would be possible to implement p.Kill() such that it caused the wait (using method on the same process object) to complete (as tested here) but independently terminating the process (or allowing it to exit) did not. I guess that's what I was thinking.

If that was really a concern, Kill could be called on a Process.GetProcessById(p.Id) object.  But I don't think it matters right now.

**Comment by danmosemsft** [comment]
Yes it's not worth more time.

**Comment by danmosemsft** [comment]
@steveisok two of the Android tests failed thus:
```
[16:20:16] dbug: Executing command: '/home/helixbot/work/9E7008EA/p/microsoft.dotnet.xharness.cli/1.0.0-prerelease.20559.2/runtimes/any/native/adb/linux/adb  start-server'
[16:20:16] dbug: 
[16:20:16] dbug: Executing command: '/home/helixbot/work/9E7008EA/p/microsoft.dotnet.xharness.cli/1.0.0-prerelease.20559.2/runtimes/any/native/adb/linux/adb  devices -l'
[16:20:16] dbug: Unexpected response from adb devices -l:
                 Exit code=0
                 Std. Output: List of devices attached
```

**Comment by danmosemsft** [comment]
FileSystemWatcher failure is already tracked by https://github.com/dotnet/runtime/issues/24181

**Pull request merged**